src/tuf/akhttpsreposource.cc

```diff
@@ -20,13 +29,8 @@ AkHttpsRepoSource::~AkHttpsRepoSource() {
   }
 }
 
-void AkHttpsRepoSource::init(std::string name_in, boost::property_tree::ptree& pt) {
+void AkHttpsRepoSource::init(std::string name_in, boost::property_tree::ptree& pt, Config& config) {
   name_ = name_in;
-
-  boost::program_options::variables_map m;
-  Config config(m);
-  fillConfig(config, pt);
-
   // Libaktualizr places certificate files inside a sqlite database
   tmp_dir_path_ = boost::filesystem::temp_directory_path() /
```

Comment:
Why do we need this tmp dir? IIRC, there is some redundancy in aklite with it.
If key/cert are stored in  HSM then there is no any tmp dir/files are required at all since libcurl/ssl loads them directly from HSM.
In the no HSM case, a key and certs are already on a file system, why do we need to copy them to some tmp storage?

Reply:
I'll try to call `http_client->setCerts` directly, that might remove the need for the tmp dir. Working on it.

Reply:
I've update the code to use `http_client->setCerts`, removing the use of a tmp dir.

Reply:
It works in my local setup.